### PR TITLE
feat: add support for Dataset.isCaseInsensitive

### LINKED ISSUE #1670: The Dataset class lacks an is_case_insensitive property
The `Dataset` class lacks an `is_case_insensitive` property.

Most of the fields of the [Dataset REST resource](https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets#resource:-dataset) have an equivalent property inside the [Dataset class](https://github.com/googleapis/python-bigquery/blob/514d3e12e5131bd589dff08893fd89bf40338ba3/google/cloud/bigquery/dataset.py)
For example:
* description has a [getter and a setter](https://github.com/googleapis/python-bigquery/blob/514d3e12e5131bd589dff08893fd89bf40338ba3/google/cloud/bigquery/dataset.py#L743)
* labels has a [getter and a setter](https://github.com/googleapis/python-bigquery/blob/514d3e12e5131bd589dff08893fd89bf40338ba3/google/cloud/bigquery/dataset.py#L723)
* defaultEncryptionConfig has a [getter and a setter](https://github.com/googleapis/python-bigquery/blob/514d3e12e5131bd589dff08893fd89bf40338ba3/google/cloud/bigquery/dataset.py#L743) as `default_encryption_config`
* ...

However, the Dataset class unfortunately doesn't have an `is_case_insensitive` property (for the `isCaseInsensitive` field), and this property is needed to configure a dataset in some use cases. I plan on creating a PR to fix this.

### PR DESCRIPTION
**Pull request by joseignaciorc**
This commits add the the `is_case_insensitive` property to the `Dataset` class. 

This property uses the `isCaseInsensitive` field of the [Dataset REST API](https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets#resource:-dataset), allowing the configuration of the case-sensitivity of the BigQuery datasets.

Fixes: https://github.com/googleapis/python-bigquery/issues/1670

### CODE REVIEW

File: google/cloud/bigquery/dataset.py [hunk 1]
```diff
@@ -764,6 +765,25 @@ def default_encryption_configuration(self, value):
             api_repr = value.to_api_repr()
         self._properties["defaultEncryptionConfiguration"] = api_repr
 
+    @property
+    def is_case_insensitive(self):
+        """Optional[bool]: TRUE if the dataset and its table names are case-insensitive, otherwise FALSE.
```
Comment (Linchin) [inline google/cloud/bigquery/dataset.py:770]: nit: use `True` and `False`, following Python convention

File: tests/system/test_client.py [hunk 2]
```diff
@@ -335,6 +349,46 @@ def test_create_table(self):
...
+        self.assertTrue(_table_exists(table_arg))
+        self.assertTrue(_table_exists(tablemc_arg))
+        self.assertIs(ci_dataset.is_case_insensitive, True)
+
+    def test_create_tables_in_case_sensitive_dataset(self):
+        ci_dataset = self.temp_dataset(
+            _make_dataset_id("create_table"), is_case_insensitive=False
+        )
+        table_arg = Table(ci_dataset.table("test_table3"), schema=SCHEMA)
+        tablemc_arg = Table(ci_dataset.table("Test_taBLe3"))
```
Comment (Linchin) [inline tests/system/test_client.py:371]: another nit: what does `mc` stand for in the variable `tablemc_arg`? Could we use something easier to understand?

**Comment by Linchin** [comment]
The PR seems good to me except for a few nits. cc @chalmerlowe to double check.

File: google/cloud/bigquery/dataset.py (hunk 1 above)
Comment (joseignaciorc) [reply google/cloud/bigquery/dataset.py]: Changed to `True` and `False`

File: tests/system/test_client.py (hunk 2 above)
Comment (joseignaciorc) [reply tests/system/test_client.py]: `mc` = mixed case; added a comment about that

**Comment by joseignaciorc** [comment]
Ping @Linchin & @chalmerlowe :)

**Comment by chalmerlowe** [comment]
@joseignaciorc 

I have been out with COVID. Still recovering and trying to catch up on all the things. I will look at this as soon as possible. Thanks for being patient.

File: tests/system/test_client.py [hunk 3]
```diff
@@ -2298,6 +2354,7 @@ def test_nested_table_to_arrow(self):
         self.assertTrue(pyarrow.types.is_list(record_col[1].type))
         self.assertTrue(pyarrow.types.is_int64(record_col[1].type.value_type))
 
+
```
Comment (Linchin) [inline tests/system/test_client.py:2357]:
```suggestion

```

**Comment by Linchin** [comment]
Thank you @joseignaciorc!

**Review by Linchin — APPROVED** [review]

**Pull request merged**